File: stonesOnTable.js
const stonesOnTable = ( n, s ) =>
{
    let counter=0;

    for ( let i = 0; i < n; i++ )
    {
        if ( s[ i ] === s[ i - 1 ] )
        {
            counter++
        }

        console.log(s[i], s[i-1])
    }

    console.log(counter)
}

stonesOnTable( 4, "abcd" );

File: stonesOnTable.c
#include <stdio.h>

int main() {
    int n;
    scanf("%d", &n);
    
    char prev, curr;
    int counter = 0;

    scanf(" %c", &prev); 
    
    for (int i = 1; i < n; i++) {
        scanf(" %c", &curr);
        
        if (curr == prev) {
            counter++;
        }
        
        prev = curr;
    }

    printf("%d\n", counter);

    return 0;
}